users/src/main/java/com/tekcamp/users/Services/Implementations/UserServiceImplementation.java

```diff
@@ -0,0 +1,115 @@
+package com.tekcamp.users.Services.Implementations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import com.tekcamp.users.Dto.UserDto;
+import com.tekcamp.users.Exceptions.UserNotFoundException;
+import com.tekcamp.users.Model.User;
+import com.tekcamp.users.Services.UserServices;
+import com.tekcamp.users.Shared.Utils;
+import com.tekcamp.users.dao.UserRepository;
+
+@Service
+public class UserServiceImplementation implements UserServices {
+
+	private UserRepository userRepository;
+	private Utils utils; 
+
+	public UserServiceImplementation(UserRepository userRepository, Utils utils) {
+		this.userRepository = userRepository;
+		this.utils = utils; 
+	}
+
+	@Override
+	public List<UserDto> getAllUsers(int page, int limit) {
+		
+		List<User> userList = new ArrayList<User>(); 
+		
+		if (page>0) page --; 
+		PageRequest pageableRequest = PageRequest.of(page, limit); 
+		
+		Page<User> userPageList = userRepository.findAll(pageableRequest);
+		
+		userList = userPageList.getContent(); 
+		
+		
+		List<UserDto> userDtoList = new ArrayList<UserDto>(); 
+		
+		for ( int i = 0; i<userList.size(); i++ ) {
+			UserDto userDto = new UserDto(); 
+			BeanUtils.copyProperties(userList.get(i), userDto);
+			userDtoList.add(userDto); 
+		}
+		
+		
+		return userDtoList;
+	}
+
+	@Override
+	public UserDto getUserByUserId(String userId) {
+		UserDto returnValue = new UserDto(); 
+		User user;
```

Comment:
just put this on line 61 where you initialize it...I'm being nitpicky because I think it's fair to show you where you can improve too, even though you did a fantastic job